emitter/gkelog/traceextractors/otel/span_extractor_test.go

```diff
@@ -0,0 +1,55 @@
+package otel
+
+import (
+	"context"
+	"encoding/hex"
+	"testing"
+
+	"github.com/vimeo/alog/v3/emitter/gkelog"
+	"go.opentelemetry.io/otel/api/trace/testtrace"
+)
+
+func TestExtractSpanInfo(t *testing.T) {
+	outerCtx := context.Background()
+
+	tracer := testtrace.NewTracer()
+
+	for _, itbl := range []struct {
+		name   string
+		ctxGen func() (context.Context, gkelog.SpanContext)
+	}{
+		{
+			name: "nospan",
+			ctxGen: func() (context.Context, gkelog.SpanContext) {
+				return outerCtx, gkelog.SpanContext{}
+			},
+		},
+		{
+			name: "no_sample_span",
+			ctxGen: func() (context.Context, gkelog.SpanContext) {
+				ctx, unsampledspan := tracer.Start(outerCtx, "foobar")
+				sctx := unsampledspan.SpanContext()
+
+				return ctx, gkelog.SpanContext{
+					SpanID:  hex.EncodeToString(sctx.SpanID[:]),
+					TraceID: hex.EncodeToString(sctx.TraceID[:]),
```

Comment:
why not use the `String()` functions for those?

Reply:
It was a misplaced desire to avoid having identical code in the test and tested code. (which ended up relying on an implementation detail of the library in use)

I've changed it.